src/com/shashi/beans/Discount.java

```diff
@@ -0,0 +1,125 @@
+package com.shashi.beans;
+
+import java.time.LocalDate;
+import java.time.Period;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import com.shashi.utility.DBUtil;
+
+public class Discount {
+    private String productId;
+    private double discountedPrice;
+    private LocalDate startDate;
+    private LocalDate endDate;
+
+    private double salePercentage;
+
+
+    // Constructors, getters, and setters
+
+    // Default constructor
+    public Discount() { }
+
+    public Discount(String productId, double discountedPrice, LocalDate startDate, LocalDate endDate) {
+        this.productId = productId;
+        this.discountedPrice = discountedPrice;
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public String getProductId() {
+        return productId;
+    }
+
+    public void setProductId(String productId) {
+        this.productId = productId;
+    }
+
+    public double getDiscountedPrice() {
+        return discountedPrice;
+    }
+
+    public void setDiscountedPrice(double discountedPrice) {
+        this.discountedPrice = discountedPrice;
+    }
+
+    public LocalDate getStartDate() {
+        return startDate;
+    }
+
+    public void setStartDate(LocalDate startDate) {
+        this.startDate = startDate;
+    }
+
+    public LocalDate getEndDate() {
+        return endDate;
+    }
+
+    public void setEndDate(LocalDate endDate) {
+        this.endDate = endDate;
+    }
+
+    public double getSalePercentage() {
+        return salePercentage;
+    }
+
+    public void setSalePercentage(double salePercentage) {
+        this.salePercentage = salePercentage;
+
+        // If sale percentage is zero, remove the discount information from the database
+        if (salePercentage == 0) {
+            removeDiscountFromDatabase();
+        }
+    }
+
+    private void removeDiscountFromDatabase() {
```

Comment:
This should be in a DiscountService class, similar to ProductService. we want each class to have its own task